### PR TITLE
[skip gpuci] Feat/cibuildwheel

### PR DESCRIPTION
**Pull request by sevagh**
Build pip wheels from GitHub Actions.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -166,3 +166,7 @@ dask-worker-space/
 # Sphinx docs & build artifacts
 docs/cudf/source/api_docs/generated/*
 docs/cudf/source/api_docs/api/*
+
+# cibuildwheel
+/wheelhouse
+/dist
```
Comment (bdice) [inline .gitignore:172]: `dist/` is already listed on line 22. We probably don't need it twice.

File: cpp/CMakeLists.txt [hunk 2]
```diff
@@ -48,6 +48,7 @@ option(BUILD_TESTS "Configure CMake to build tests" ON)
 option(BUILD_BENCHMARKS "Configure CMake to build (google & nvbench) benchmarks" OFF)
 option(BUILD_SHARED_LIBS "Build cuDF shared libraries" ON)
 option(JITIFY_USE_CACHE "Use a file cache for JIT compiled kernels" ON)
+option(CUDF_BUILD_WHEELS "Whether we're building libcudf to go in a wheel for pypi" OFF)
```
Comment (bdice) [inline cpp/CMakeLists.txt:51]: Avoid "we". Also no need to specify PyPI here.
```suggestion
option(CUDF_BUILD_WHEELS "Whether libcudf is being built for a Python wheel" OFF)
```

File: python/cudf/CMakeLists.txt [hunk 3]
```diff
@@ -14,10 +14,22 @@
...
+  # Make FindPython{,2,3} modules use LOCATION for lookup strategy.
+  # https://cmake.org/cmake/help/latest/policy/CMP0094.html#policy:CMP0094
+  cmake_policy(SET CMP0094 NEW)
+  set(CMAKE_POLICY_DEFAULT_CMP0094 NEW)
+endif()
 
 include(../../fetch_rapids.cmake)
 
+include(rapids-cuda)
+# TODO: This will not be necessary once we upgrade to CMake 3.22, which will pull in the required
```
Comment (bdice) [inline python/cudf/CMakeLists.txt:27]: This should no longer be needed. https://github.com/rapidsai/cudf/pull/11751 requires CMake 3.23.1 or newer.

File: python/cudf/CMakeLists.txt [hunk 4]
```diff
@@ -26,40 +38,73 @@ project(
             # various linking options for the python library is hardcoded to build with C, so until
             # that is fixed we need to keep C.
             C CXX
+            # TODO: This will not be necessary once we upgrade to CMake 3.22, which will pull in the required
```
Comment (bdice) [inline python/cudf/CMakeLists.txt:41]: Similarly, this workaround can be removed because https://github.com/rapidsai/cudf/pull/11751 is merged.

**Review by bdice — COMMENTED** [review]
A few comments - I was interested in how this code works. Pretty awesome!

**Comment by vyasr** [comment]
Superseded by #12096

**Pull request closed without merging**